### ISSUE TITLE
Spurious warning log from IstioOperator

### ISSUE THREAD

**Issue by esnible** [istio/istio#21519]
When I use the operator to install Istio, all of the logs are at INFO level except one:

```
2020-02-26T15:42:27.994294Z	warn	retrieving resources to prune type certmanager.k8s.io/v1beta1, Kind=ClusterIssuer: no matches for kind "ClusterIssuer" in version "certmanager.k8s.io/v1beta1" not found
```
Everything seems fine and the components went into 3 = HEALTHY status.

To reproduce: I followed the instructions at https://preliminary.istio.io/docs/setup/install/standalone-operator/ .  I already had a running Istio at that time.

**Comment by morvencao** [comment]
This is because the prune list is static so far, which means the controller will alway try to prune stale resource from prune list. https://github.com/istio/istio/blob/master/operator/pkg/controller/istiocontrolplane/pruningdetails.go

**Comment by howardjohn** [comment]
We have fixed this #23314
